bin/verify-binary-architecture.sh

```diff
@@ -0,0 +1,47 @@
+#!/usr/bin/env bash
+set -xeuo pipefail
+
+if [ $# -ne 2 ]; then
+  echo "Usage: $0 <target-triple> <binary-path>"
+  exit 1
+fi
+TARGET_TRIPLE="$1"
+BINARY_PATH="$2"
+BINARY_FILE_TYPE=$(file -b "$BINARY_PATH")
+
+# Architecture patterns
+ARCHITECTURE_PATTERNS_FOR_AARCH64='aarch64|Aarch64|arm64'
+ARCHITECTURE_PATTERNS_FOR_X86_64='x86-64|x86_64'
+
+# Architecture helper functions
+parse_architecture() {
+  grep -Eo \
+    -e "$ARCHITECTURE_PATTERNS_FOR_AARCH64" \
+    -e "$ARCHITECTURE_PATTERNS_FOR_X86_64" \
+    | head -n1
+}
+normalize_architecture() {
+  sed -E \
+    -e "s/($ARCHITECTURE_PATTERNS_FOR_AARCH64)/AArch64/" \
+    -e "s/($ARCHITECTURE_PATTERNS_FOR_X86_64)/x86-64/"
+}
```

Comment:
Please remove all the logic from this script relating to converting strings to other strings. I want this script to accept an exactly string and check that the output of file matches it. There should be zero mapping logic in here at all, it is too hard to understand.